blockchain/src/blockchain/push.rs

```diff
@@ -307,10 +307,22 @@ impl Blockchain {
         this.chain_store.set_head(&mut txn, &block_hash);
 
         if is_election_block {
-            this.chain_store.prune_epoch(
-                Policy::epoch_at(block_number).saturating_sub(MAX_EPOCHS_STORED),
-                &mut txn,
-            );
+            let max_epochs_stored =
+                cmp::max(this.config.max_epochs_stored, Policy::MIN_EPOCHS_STORED);
+
+            // Calculate the epoch to be pruned. Saturate at zero.
+            let pruned_epoch = Policy::epoch_at(block_number).saturating_sub(max_epochs_stored);
+
+            if this.config.keep_history {
+                // If we are a history node, we only prune the chain store
+                this.chain_store.prune_epoch(pruned_epoch, &mut txn);
+            } else {
+                // Prune the History Store.
+                this.history_store.remove_history(&mut txn, pruned_epoch);
```

Comment:
```suggestion
                this.history_store.remove_history(&mut txn, Policy::epoch_at(block_number).saturating_sub(1));
```
We don't want to keep history at all for full node